zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -336,88 +345,82 @@ mod test {
     }
 
     fn mock_tx(
-        value: Option<Word>,
-        gas: Option<u64>,
-        gas_price: Option<Word>,
+        value: Word,
+        gas: Word,
+        gas_price: Word,
         calldata: Vec<u8>,
     ) -> eth_types::Transaction {
         let from = address!("0x00000000000000000000000000000000000000fe");
         let to = address!("0x00000000000000000000000000000000000000ff");
-        let minimal_gas =
-            Word::from(GasCost::TX.as_u64() + 2 * OpcodeId::PUSH32.constant_gas_cost().as_u64());
-        let point_one_ether = Word::from(10u64.pow(17));
-        let two_gwei = Word::from(2_000_000_000);
         eth_types::Transaction {
             from,
             to: Some(to),
-            value: value.unwrap_or(point_one_ether),
-            gas: gas.map(Word::from).unwrap_or(minimal_gas),
-            gas_price: gas_price.or(Some(two_gwei)),
+            value,
+            gas,
+            gas_price: Some(gas_price),
             input: calldata.into(),
             ..Default::default()
         }
     }
 
     #[test]
     fn begin_tx_gadget_simple() {
-        // Transfer 1 ether, successfully
-        test_ok(mock_tx(None, None, None, vec![]), true);
+        // Transfer 0.1 ether, successfully
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            true,
+        );
 
-        // Transfer 1 ether, tx reverts
-        test_ok(mock_tx(None, None, None, vec![]), false);
+        // Transfer 0.1 ether, tx reverts
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            false,
+        );
 
         // Transfer nothing with some calldata
         test_ok(
-            mock_tx(None, Some(21086), None, vec![1, 2, 3, 4, 0, 0, 0, 0]),
+            mock_tx(
+                Word::zero(),
+                Word::from(21086),
+                *TWO_GWEI,
+                vec![1, 2, 3, 4, 0, 0, 0, 0],
+            ),
             false,
         );
     }
 
     #[test]
     fn begin_tx_gadget_rand() {
-        let point_one_ether = Word::from(10u8).pow(Word::from(17u8));
+        let random_amount = Word::from_little_endian(&rand_bytes(32)) % *POINT_ONE_ETHER;
+
+        let max_gas_price = 476_054_460_630_296u64;
+        assert_eq!(Word::from(max_gas_price), *TEN_ETHER / *MINIMAL_GAS);
+        let random_gas_price = Word::from(rand_range(0..max_gas_price));
```

Comment:
We should move this too.

Reply:
I added a function to compute this directly instead of using hard coded values.